client/src/components/openlayersMap.tsx

```diff
@@ -0,0 +1,115 @@
+import React, {useState, useEffect, useRef} from 'react';
+import Map from 'ol/Map';
+import View from 'ol/View';
+import Feature, {FeatureLike} from 'ol/Feature';
+import Geometry from 'ol/geom/Geometry';
+import VectorLayer from 'ol/layer/Vector';
+import VectorSource from 'ol/source/Vector';
+import {fromLonLat} from 'ol/proj';
+import {Coordinate} from 'ol/coordinate';
+import olms from 'ol-mapbox-style';
+import mapStyle from '../data/mapStyle';
+import ZoomWarning from './zoomWarning';
+import MapPopup from './mapPopup';
+import * as styles from './openlayersMap.module.scss';
+
+interface IMapWrapperProps {
+  features: Feature<Geometry>[],
+};
+
+// The below adapted from
+// https://taylor.callsen.me/using-openlayers-with-react-functional-components/
+const MapWrapper = ({features}: IMapWrapperProps) => {
+  const [map, setMap] = useState<Map>();
+  const [featuresLayer, setFeaturesLayer] = useState<VectorLayer>();
+  const [selectedFeature, setSelectedFeature] = useState<FeatureLike>();
+  const [currentZoom, setCurrentZoom] = useState<number>(4);
+  const [currentOverlayPosition, setCurrentOverlayPosition] = useState<Coordinate>([]);
+
+  const mapElement = useRef() as
+    React.MutableRefObject<HTMLInputElement>;
+
+  // create state ref that can be accessed in OpenLayers onclick callback function
+  //  https://stackoverflow.com/a/60643670
+  const mapRef = useRef() as React.MutableRefObject<Map>;
+  if (map) {
+    mapRef.current = map;
+  }
+
+  useEffect( () => {
+    const view = new View({
+      center: fromLonLat([-86.502136, 32.4687126]),
```

Comment:
Maybe move to named const at some point so we know where/why this is lat/lng magic number?